terraform/main.tf

```diff
@@ -0,0 +1,61 @@
+# Enable Google Cloud APIs
+module "enable_google_apis" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "~> 13.0"
+
+  project_id                  = var.gcp_project_id
+  disable_services_on_destroy = false
+
+  activate_apis               = var.apis
+}
+
+# Create GKE Cluster
+resource "google_container_cluster" "cluster" {
+  name               = var.name
+  location           = var.region
+  ip_allocation_policy {
```

Comment:
is there a reason why this is still in if empty? I see it as an optional config in the [hashicorp](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster) doc

Reply:
@ckim328 this is a workaround for a bug that I was running into with using autopilot cluster. The issue (and the workaround suggestion in the comments) can be found here: https://github.com/hashicorp/terraform-provider-google/issues/10782

Reply:
@jaspermai in such event, please add a comment that reasons the usage.
Why the selected solution is not [using the version of the gcp](https://github.com/hashicorp/terraform-provider-google/issues/10782#issuecomment-1024047928) provided that does not have the issue?
Also, can you confirm that you tried it with the latest terraform (1.2.5)?

Reply:
Thanks Leonid, I'll take a look into specifying the version of gcp. I have tested with the latest Terraform (1.2.5) version and the issue was still there.